ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TopicInfoCache.java

```diff
@@ -87,28 +155,34 @@ public void clear() {
 
   private TopicInfo load(final String topicName) {
     try {
-      final java.util.regex.Matcher matcher = INTERNAL_TOPIC_PATTERN.matcher(topicName);
-      if (matcher.matches()) {
+      final Optional<InternalTopic> internalTopic = INTERNAL_TOPIC_PATTERNS.stream()
+          .map(p -> p.match(topicName))
+          .filter(Optional::isPresent)
+          .map(Optional::get)
+          .findFirst();
+
+      if (internalTopic.isPresent()) {
         // Internal topic:
-        final QueryId queryId = new QueryId(matcher.group(1));
         final PersistentQueryMetadata query = ksqlEngine
-            .getPersistentQuery(queryId)
+            .getPersistentQuery(internalTopic.get().queryId())
             .orElseThrow(() -> new TestFrameworkException("Unknown queryId for internal topic: "
-                + queryId));
+                + internalTopic.get().queryId())
+            );
 
         final java.util.regex.Matcher windowedJoinMatcher = WINDOWED_JOIN_PATTERN
             .matcher(query.getStatementString());
 
         final OptionalLong changeLogWindowSize = topicName.endsWith("-changelog")
             && windowedJoinMatcher.matches()
-            ? OptionalLong.of(DurationParser.parse(windowedJoinMatcher.group(1)).toMillis())
+            ? OptionalLong
+            .of(DurationParser.parse(windowedJoinMatcher.group("duration")).toMillis())
             : OptionalLong.empty();
 
         return new TopicInfo(
             topicName,
             query.getLogicalSchema(),
-            query.getResultTopic().getKeyFormat(),
-            query.getResultTopic().getValueFormat(),
```

Comment:
This is the bad code. It was getting the internal topic's key and value format from the sink topic... which just ain't right ;)